File: CS/Helpers/MDIBehavior.cs
using DevExpress.Mvvm.UI;
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Diagram;
using DevExpress.Xpf.Docking;
using System.Linq;

namespace MDI_Diagram {
    public class MDIBehavior : Behavior<DiagramControl> {

        void Init() {
            var panel = LayoutTreeHelper.GetVisualParents(AssociatedObject).OfType<LayoutPanel>().FirstOrDefault();
            if (panel != null)
                DiagramControl.SetDiagram(panel, AssociatedObject);
        }

        protected override void OnAttached() {
            base.OnAttached();

            if (AssociatedObject.IsLoaded)
                Init();
            else
                AssociatedObject.Loaded += (s, e) => {
                    Init();
                };
        }
    }
}

File: VB/Helpers/MDIBehavior.vb
Imports DevExpress.Mvvm.UI
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Diagram
Imports System.Linq

Namespace MDI_Diagram

    Public Class MDIBehavior
        Inherits Behavior(Of DiagramControl)

        Private Sub Init()
            Dim panel = LayoutTreeHelper.GetVisualParents(AssociatedObject).OfType(Of LayoutPanel)().FirstOrDefault()
            If panel IsNot Nothing Then DiagramControl.SetDiagram(panel, AssociatedObject)
        End Sub

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            If AssociatedObject.IsLoaded Then
                Init()
            Else
                AssociatedObject.Loaded += Function(s, e)
                    Init()
                End Function
            End If
        End Sub
    End Class
End Namespace
